pkg/ssl.go

```diff
@@ -0,0 +1,204 @@
+package pkg
+
+import (
+	"encoding/pem"
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"github.com/wttech/aemc/pkg/common/cryptox"
+	"github.com/wttech/aemc/pkg/common/filex"
+	"github.com/wttech/aemc/pkg/common/osx"
+	"github.com/wttech/aemc/pkg/common/pathx"
+	"golang.org/x/net/html"
+	"io"
+	"os"
+	"strings"
+)
+
+const (
+	SSLSetupPath = "/libs/granite/security/post/sslSetup.html"
+)
+
+type SSL struct {
+	instance *Instance
+}
+
+type sslLock struct {
+	KeystorePassword   string `yaml:"keystore_password"`
+	TrustStorePassword string `yaml:"trust_store_password"`
+	Certificate        string `yaml:"certificate"`
+	PrivateKey         string `yaml:"private_key"`
+	HttpsHostname      string `yaml:"https_hostname"`
+	HttpsPort          string `yaml:"https_port"`
+}
+
+func NewSSL(instance *Instance) *SSL {
+	return &SSL{
+		instance: instance,
+	}
+}
+
+func (s SSL) Setup(keyStorePassword, trustStorePassword, certificateFile, privateKeyFile, httpsHostname, httpsPort string) (bool, error) {
+	if !pathx.Exists(certificateFile) {
+		return false, fmt.Errorf("%s > certificate file does not exist: %s", s.instance.ID(), certificateFile)
```

Comment:
`SSL certificate`...

Reply:
imagine that you see that error somewhere and don't know the context... SSL will be helpful that it is connected to SSL ...not to Git, SSH, or whatever else